go/arrow/compute/internal/exec/kernel.go

```diff
@@ -242,6 +242,40 @@ func (primitiveMatcher) Equals(other TypeMatcher) bool {
 // returns true for.
 func Primitive() TypeMatcher { return primitiveMatcher{} }
 
+type reeMatcher struct {
+	runEndsMatcher TypeMatcher
+	encodedMatcher TypeMatcher
+}
+
+func (r reeMatcher) Matches(typ arrow.DataType) bool {
+	if typ.ID() != arrow.RUN_END_ENCODED {
+		return false
+	}
+
+	dt := typ.(*arrow.RunEndEncodedType)
+	return r.runEndsMatcher.Matches(dt.RunEnds()) && r.encodedMatcher.Matches(dt.Encoded())
+}
+
+func (r reeMatcher) Equals(other TypeMatcher) bool {
+	o, ok := other.(reeMatcher)
+	if !ok {
+		return false
+	}
+	return r.runEndsMatcher.Equals(o.runEndsMatcher) && r.encodedMatcher.Equals(o.encodedMatcher)
+}
+
+func (r reeMatcher) String() string {
+	return "run_length_encoded(" + r.encodedMatcher.String() + ")"
```

Comment:
should runEndsMatcher be included in the repr?

Reply:
good point, it should. i'll fix that